beta-src/src/components/map/components/WDFlyoutContainer.tsx

```diff
@@ -19,80 +17,113 @@ interface WDFlyoutContainerProps {
   units: Unit[];
 }
 
-const WDFlyoutContainer: React.FC<WDFlyoutContainerProps> = function ({
+const WDFlyoutContainer: FC<WDFlyoutContainerProps> = function ({
   units,
-}): React.ReactElement {
+}): ReactElement {
   const dispatch = useAppDispatch();
   const order = useAppSelector(gameOrder);
   const maps = useAppSelector(gameMaps);
   const legalOrders = useAppSelector(gameLegalOrders);
 
-  // console.log("FLYOUT");
-  // console.log({ order });
+  const unit = units.find((u) => u.unit.id === order.unitID);
+  const mTerr = TerritoryMap[maps.unitToTerritory[order.unitID]];
+
+  const clickHandler = (
+    orderType,
+    viaConvoy: string | undefined = undefined,
+  ) => {
+    if (!order.inProgress) return;
+    console.log(`Dispatched ${orderType}`);
+    dispatch(
+      gameApiSliceActions.updateOrder({
+        type: orderType,
+        viaConvoy,
+      }),
+    );
+  };
+
+  const canConvoy: boolean =
+    unit?.unit?.type === "Fleet" &&
+    mTerr.provinceMapData.type === "Sea" &&
+    legalOrders.hasAnyLegalConvoysByUnitID[order.unitID];
+
+  const fleetAndSeaAction = () => {
+    if (!order.inProgress) return;
+
+    if (canConvoy) {
+      clickHandler("Convoy");
+    }
+  };
+
+  const canVia: boolean =
+    unit?.unit?.type === "Army" &&
+    legalOrders.legalViasByUnitID[order.unitID].length > 0;
+
+  const moveAction = () => {
+    if (!order.inProgress) return;
+
+    if (canVia) {
+      clickHandler("Move", "Yes");
+    } else {
+      clickHandler("Move");
+    }
+  };
+
+  useKeyPressEvent("h", () => clickHandler("Hold"));
+  useKeyPressEvent("d", () => clickHandler("Hold"));
+  useKeyPressEvent("m", moveAction);
+  useKeyPressEvent("a", moveAction);
+  useKeyPressEvent("v", moveAction);
+  useKeyPressEvent("s", () => clickHandler("Support"));
+  useKeyPressEvent("c", fleetAndSeaAction);
```

Comment:
Here's explicitly what I think this should look like to give correct behavior for vias.

```suggestion
  const canVia: boolean =
    unit?.unit?.type === "Army" &&
    legalOrders.legalViasByUnitID[order.unitID].length > 0;
  };

  useKeyPressEvent("h", () => clickHandler("Hold"));
  useKeyPressEvent("d", () => clickHandler("Hold"));
  useKeyPressEvent("m", () => clickHandler("Move"));
  useKeyPressEvent("a", () => clickHandler("Move"));
  useKeyPressEvent("v", () => canVia && clickHandler("Move", "Yes"));
  useKeyPressEvent("s", () => clickHandler("Support"));
  useKeyPressEvent("c", () => clickHandler("Convoy"));
```